assets/js/googlesitekit/data/create-fetch-store.test.js

```diff
@@ -0,0 +1,328 @@
+	controlCallback: ( params ) => {
+		const { aParam, objParam } = params;
+		return API.get( 'core', 'test', 'some-data', {
+			aParam,
+			objParam,
+		} );
+	},
+	reducerCallback: ( state, response, params ) => {
+		const { aParam } = params;
+		return {
+			...state,
+			data: {
+				...( state.data || {} ),
+				[ aParam ]: response,
+			},
+		};
+	},
+	argsToParams: ( objParam, aParam ) => {
+		invariant( 'object' === typeof objParam, 'objParam is required.' );
+		invariant( 'undefined' !== typeof aParam, 'aParam is required.' );
+		return {
+			objParam,
+			aParam,
+		};
+	},
+};
+
+describe( 'createFetchStore store', () => {
+	let apiFetchSpy;
+	let dispatch;
+	let registry;
+	let select;
+	let storeDefinition;
+	let store;
+
+	beforeAll( () => {
+		API.setUsingCache( false );
+	} );
+
+	beforeEach( () => {
+		registry = createRegistry();
+
+		storeDefinition = createFetchStore( STORE_PARAMS );
+		registry.registerStore( STORE_NAME, storeDefinition );
+		dispatch = registry.dispatch( STORE_NAME );
+		store = registry.stores[ STORE_NAME ].store;
+		select = registry.select( STORE_NAME );
+
+		apiFetchSpy = jest.spyOn( { apiFetch }, 'apiFetch' );
+	} );
+
+	afterAll( () => {
+		API.setUsingCache( true );
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+		apiFetchSpy.mockRestore();
+	} );
+
+	describe( 'actions', () => {
+		it( 'includes the expected actions', () => {
+			const fetchStoreDefinition = createFetchStore( {
+				baseName: 'SaveSomeData',
+				controlCallback: async () => true,
+				reducerCallback: ( state ) => state,
+			} );
+
+			expect( Object.keys( fetchStoreDefinition.actions ) ).toEqual( [
+				'fetchSaveSomeData',
+				'receiveSaveSomeData',
+			] );
+		} );
+
+		describe( 'fetch', () => {
+			it( 'validates parameters based on argsToParams', () => {
+				const consoleErrorSpy = jest.spyOn( global.console, 'error' );
+
+				dispatch.fetchGetSomeData();
+				expect( consoleErrorSpy ).toHaveBeenCalledWith( 'objParam is required.' );
+
+				dispatch.fetchGetSomeData( 123 );
+				expect( consoleErrorSpy ).toHaveBeenCalledWith( 'objParam is required.' );
+
+				dispatch.fetchGetSomeData( {} );
+				expect( consoleErrorSpy ).toHaveBeenCalledWith( 'aParam is required.' );
+
+				consoleErrorSpy.mockClear();
+			} );
+
+			it( 'yields the expected actions for an arguments error', () => {
+				const fetchStoreDefinition = createFetchStore( {
+					baseName: 'SaveSomeData',
+					controlCallback: async () => true,
+					reducerCallback: ( state ) => state,
+					argsToParams: ( requiredParam ) => {
+						invariant( requiredParam, 'requiredParam is required.' );
+						return {
+							requiredParam,
+						};
+					},
+				} );
+
+				const action = fetchStoreDefinition.actions.fetchSaveSomeData();
+
+				// Catch invariant to get exactly the error we expect.
+				let error;
+				try {
+					invariant( false, 'requiredParam is required.' );
+				} catch ( err ) {
+					error = err;
+				}
```

Comment:
It would be good to extract this to a utility as I would imagine this is probably used more than once in these tests.

I'm curious though, would it not pass with simply `new Error( 'requiredParam is required.' )` ?

Reply:
Unfortunately not, the errors that `invariant` throws have a different format.